internal/helpers/hostname_source.go

```diff
@@ -0,0 +1,138 @@
+package helpers // import "go.opentelemetry.io/ebpf-profiler/internal/helpers"
```

Comment:
Nit: I'd rename this to `address.go` as all functions are related to network addresses.

Reply:
Sounds good. I did have an hesitation with this filename.